### ISSUE TITLE
Could games-on-whales support WebRTC to stream to a browser?

### ISSUE THREAD

**Issue by guozanhua218** [games-on-whales/wolf#78]

**Comment by ABeltramo** [comment]
At the moment Wolf only supports Moonlight but, since most of the components that are used here are completely independent it's not too hard to add support for other types of streaming protocols.  

Gstreamer (the library that we use in order to create the audio/video pipeline) has great support for WebRTC out of the box. The only real bit missing to make this happening is "just" a proper Web UI and some glue code to expose the client devices (mouse, keyboard, joypads) over the browser and all the way back to the remote Wolf instance.

This is something that I might revisit in future, if anyone wants to give it a shot I'd be happy to help and provide all the relevant information in how to integrate WebRTC (or even better MoQ).

**Comment by guozanhua218** [comment]
> At the moment Wolf only supports Moonlight but, since most of the components that are used here are completely independent it's not too hard to add support for other types of streaming protocols.
> 
> Gstreamer (the library that we use in order to create the audio/video pipeline) has great support for WebRTC out of the box. The only real bit missing to make this happening is "just" a proper Web UI and some glue code to expose the client devices (mouse, keyboard, joypads) over the browser and all the way back to the remote Wolf instance.
> 
> This is something that I might revisit in future, if anyone wants to give it a shot I'd be happy to help and provide all the relevant information in how to integrate WebRTC (or even better MoQ).

Yes, I am interested in this. Can you provide me with more information in how to integrate WebRTC(or MoQ) and then I will try

**Comment by ABeltramo** [comment]
Sure, I've put down some basic instructions in a gist [here](https://gist.github.com/ABeltramo/39284b1731bb1aea4c09d425ec0fcd6a). That should get you started with `gst-wayland-display` + `inputtino` so that you'll hopefully end up with something like:

https://github.com/games-on-whales/wolf/assets/1122217/7be6db9c-5216-4b03-b4bd-1c9075d9e124

I hope this helps as a very basic starting point, let me know how it goes!

**Comment by guozanhua218** [comment]
Current the solution of GOW  only support the games in linux platform, can't support window's games?

**Comment by ABeltramo** [comment]
> Current the solution of GOW  only support the games in linux platform, can't support window's games?

Thanks to Steam Proton, we can run most Windows games in Linux!

**Comment by guozanhua218** [comment]
> > Current the solution of GOW  only support the games in linux platform, can't support window's games?
> 
> Thanks to Steam Proton, we can run most Windows games in Linux!

Use wine in Linux?

**Comment by guozanhua218** [comment]
> > > Current the solution of GOW  only support the games in linux platform, can't support window's games?
> > 
> > 
> > Thanks to Steam Proton, we can run most Windows games in Linux!
> 
> Use wine in Linux?

Steam Play?

**Comment by zslliu** [comment]
> Sure, I've put down some basic instructions in a gist [here](https://gist.github.com/ABeltramo/39284b1731bb1aea4c09d425ec0fcd6a). That should get you started with `gst-wayland-display` + `inputtino` so that you'll hopefully end up with something like:当然，我在这里列出了一些基本说明。这应该让你从 `gst-wayland-display` + `inputtino` 开始，这样你就有希望最终得到类似的结果：
> 
>  webrtc+inputtino.mp4 
> I hope this helps as a very basic starting point, let me know how it goes!我希望这能作为一个非常基本的起点有所帮助，让我知道进展如何！

The test connection was successful on an Ubuntu computer, but only a black window could be seen.
![2](https://github.com/games-on-whales/wolf/assets/36098473/b4f0e9a7-e499-43a4-94a4-f83a1c5f4662)
![3](https://github.com/games-on-whales/wolf/assets/36098473/20755b94-9585-4d7d-b0d4-e228365b066f)
![1](https://github.com/games-on-whales/wolf/assets/36098473/46f17bbc-f2e2-45d5-b2db-21b31f329377)

**Comment by ghost** [comment]
maybe this helps 

https://github.com/selkies-project/docker-nvidia-egl-desktop

**Comment by ehfd** [comment]
https://github.com/selkies-project/selkies-gstreamer

This fits your description.

**Comment by ABeltramo** [comment]
So does Kasm and others, this is a different project with a different focus. Besides, spamming your project in all issues is not cool.

**Comment by ehfd** [comment]
Well, the code implements exactly what the issue description does and thus is a reference. I want more implementations that can be an alternative to noVNC, not less.
